File: vihChatBot/src/main/java/com/vihmessenger/vihchatbot/data/model/GenericStatusResponse.kt
package com.vihmessenger.vihchatbot.data.model

import com.google.gson.annotations.SerializedName

/**
 * Minimal `{ status, message }` envelope returned by the enterprise state-mutation
 * endpoints (block/unblock, mute/unmute, promotional opt-in/out). All are nullable so a
 * terse backend response (e.g. status only) still decodes.
 */
data class GenericStatusResponse(
    @SerializedName("status") var status: Boolean? = null,
    @SerializedName("message") var message: String? = null
)

File: Sources/VihChatBotSDK/Models/GenericStatusResponse.swift
import Foundation

/// Minimal `{ status, message }` envelope returned by the enterprise state-mutation
/// endpoints (block/unblock, mute/unmute, promotional opt-in/out). Both fields are
/// optional so a terse backend response still decodes. Mirrors `GenericStatusResponse.kt`.
public struct GenericStatusResponse: Codable {
    public var status: Bool?
    public var message: String?
}
